src/app/login/page.tsx

```diff
@@ -39,48 +91,127 @@ const LoginPage = () => {
         login(response.token);
         router.push('/');
       }
-    } catch (error) {
-      setError('Špatný email nebo heslo.');
+    } catch (error: any) {
+      if (error.response.status === 400) {
+        setError('Špatně zadané přihlašovací údaje.');
+      } else {
+        setError('Něco se pokazilo. Zkuste to prosím znovu později.');
+      }
+      setIsLoading(false);
     }
   };
 
+  const isDisabled = Boolean(emailError || passwordError || passwordError);
+
   return (
-    <div className="flex flex-col items-center justify-center">
-      <form
-        className="mt-32 flex flex-col border border-black p-4 md:w-1/3"
-        onSubmit={handleSubmit}
-      >
-        <label className="mb-4 w-full">
-          Email:
+    <section className="flex flex-col items-center justify-center pt-20">
+      <h1 className="text-xl font-semibold">
+        Přihlášení se do <span className="text-primary">SportujSpolu</span>
+      </h1>
+      <p className="mt-4">Chci se přihlásit přes:</p>
+      <ul className="my-5 flex">
+        <li>
+          <button onClick={() => window.alert('🚀 Feature coming soon! 🌟')}>
+            <FacebookIcon className="mx-3 transform transition-transform duration-200 ease-in-out hover:scale-115 hover:fill-primary" />
+          </button>
+        </li>
+        <li>
+          <button onClick={() => window.alert('🚀 Feature coming soon! 🌟')}>
+            <GoogleIcon className="mx-3 transform transition-transform duration-200 ease-in-out hover:scale-115 hover:fill-primary" />
+          </button>
+        </li>
+      </ul>
+      <hr className="w-100 border-t border-light-gray" />
+      {!errorMessage ? (
+        <p className="pt-5">Zadat přihlašovací údaje:</p>
+      ) : (
+        <p className="mt-5 text-secondary">{errorMessage}</p>
+      )}
```

Comment:
I would not replace the headline. It could be confusing for users with readers. There is a very important accessibility standard around error messages or any pop-ups `aria-live`.

I recommended checking it out. You can actually share your findings on slack/notion so we both get smarter. The implementation will look something like this

```
import { LiveMessage } from 'react-aria-live';

export const AnnouncedNotification = (props: NotificationStatusProps) => (
  <>
    <p className="mt-5 text-secondary">{props.statusMessage}</p>
    <LiveMessage aria-live="polite" message={props.statusMessage || ''} />
  </>
);

```